### ISSUE TITLE
Feature Request: option to execute a custom command as notification

### ISSUE THREAD

**Issue by PureIncompetence** [jhonderson/gone-man-switch#3]
The ability to execute a specified command as a notification would imho greatly expand the flexibillity of this application.
The use-case I had thought about would be sending a notification using cli-tools (e.g. sendxmpp).

**Comment by jhonderson** [comment]
Can this be achieved using a webhook ? I.e you provide a url, http headers and body and the system will send a post http request using that information. 

The reason I am doubtful about the command route is 1) the user may enter malicious commands and read other users information, and 2) for docker the command will be executed inside the docket container, which won’t have installed your custom command tools (unless you modify the docker file and build your own image)

**Comment by samcro1967** [comment]
I too would be interested in a Check-in Destination that could allow me to run a script.  I would be fine with a webhook.  I already run a custom image using https://github.com/adnanh/webhook with tools in it that I need or could easily add more into the image pipeline if needed to execute the desired tasks in the script.

Would probably also use the webhook to send an email (rather than have the app do it natively) so that I can attach my file in real-time to avoid having to remember to reupload the file to the app whenever changes are made.  Could then also attach multiple files (such as pictures and our favorite song as an mp3) as an example use case.

**Comment by jhonderson** [comment]
Thanks both for your inputs!

Just to confirm the ask here, this is requesting a **webhook and/or command** as a **message destination** (i.e a destination for sending the message out when you are gone). The request is **not for supporting webhook/command for sending check-in notifications** (i.e a destination to check if you are still available). Is this correct?

**Comment by samcro1967** [comment]
That is correct.  Looking for a message destination, not a check-in notification.  A simple http (using HTTPie in the docker image perhaps) to call the webhook which in turn then would execute a script that in turn would send the email and attach files in real-time from the local host.

Might also do some other things in the script such as change the background image on the main shared household desktop, etc. 
 Would even order my wife flowers from the script if I could figure out a way to do it. :smile:

**Comment by Tharic99** [comment]
Said webhook could also be used to execute a script that wipes the entire contents of my browser history, archives critical items that I want shared with specific other individuals and purges items I do NOT want shared with others. xD

**Comment by cromelex** [comment]
Would you consider adding this feature? The ability to call  a webhook would be a great addition.
